xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/commentfield.vm

```diff
@@ -22,63 +22,41 @@
-#set ($name = $escapetool.xml($request.name))
-## the number of the comment, if no number is provided we generate a field for a new comment.
+## The number of the comment, if no number is provided we generate a field for a new comment.
 #set ($number = $numbertool.toNumber($request.number).intValue())
 
-## the 'comment' field from the XWikiComments class is used as a template for the field to display
 #set ($xCommentClass = 'XWiki.XWikiComments')
-#set ($commentClass = $xwiki.getDocument($xCommentClass).getxWikiClass())
 
-
-## the number of the comment, if no number is provided we generate a field for a new comment.
-#if ($number && $number > -1)
-  #set ($commentObj = $doc.getObject($xCommentClass, $number))
-  #set ($value = $commentObj.getValue('comment'))
-#else
-  #set ($commentObj = $doc.newObject($xCommentClass))
-  #set ($value = '')
-#end
-
-## Forces the value of the comment to the content from the request if it exists.
+## If a content is provided, we use it. If a comment number is provided, we use the content of the corresponding 
+## comment. Otherwise, the content is an empty string.
 #if ($request.content)
+  ## Forces the value of the comment to the content from the request if it exists (for instance in case of error 
+  ## during the captcha validation).
   #set ($value = $request.content)
+#elseif ($number && $number > -1)
+  #set ($value = $doc.getObject($xCommentClass, $number).getValue('comment'))
+#else
+  #set ($value = '')
 #end
 
-## We can't simply call $doc.display() on this object because we would get comment as the
-## field name. Instead we want the field name to be XWiki.XWikiComments_comment for new comments
-## and XWiki.XWikiComments_N_comment for edited comments.
-## To achieve this we add a new property to the comment class by copying the comment property and changing the name.
-#set ($commentObjClass = $commentObj.getxWikiClass())
-#set ($commentObjBaseClass = $commentObjClass.getXWikiClass())
-#set ($commentField = $commentClass.getXWikiClass().get('comment'))
-
-## In theory this should never occur, but apparently it happened in XWIKI-18717 in weird conditions,
-## so we're bullet-proofing
-#if ("$!commentField" != '')
-#set ($discard = $commentObjBaseClass.put($name, $commentField))
-#set ($discard = $commentObjBaseClass.get($name).setName($name))
-
-## attribution of the value to the new field in the object.
-#set ($discard = $commentObj.set($name, $value))
-
-#set ($fieldToDisplay = $commentObjClass.get($name))
-
 ## We need to know the type of editor used for comments in order to hide/display the preview button.
 #set ($defaultEditorId = $services.edit.syntaxContent.defaultEditorId)
 <input type='hidden' name='defaultEditorId' value="$escapetool.xml($defaultEditorId)" />
 
-<label for="$name">$services.localization.render('core.viewers.comments.add.comment.label')</label>
+#if ($number && $number > -1)
+  #set ($commentPrefix = "XWiki.XWikiComments_${number}_")
```

Comment:
Wasn't the plan to remove any explicit reference to "XWiki.XWikiComments" ?

Reply:
Not exactly. The issue title is misleading. The goal is to stop manipulating the `XWiki.XWikiComments` XClass at runtime. The current script add temporarily a field with the desired name to the XClass, to remove it once unused. Which is a terrible solution.